x-pack/legacy/plugins/file_upload/public/util/indexing_service.js

```diff
@@ -117,12 +115,13 @@ function transformDataByFormatForIndexing(transform, parsedFile, dataType) {
 }
 
 async function writeToIndex(indexingDetails) {
-  const paramString = indexingDetails.id !== undefined ? `?id=${indexingDetails.id}` : '';
+  const query = indexingDetails.id ? { id: indexingDetails.id} : null;
   const { appName, index, data, settings, mappings, ingestPipeline } = indexingDetails;
 
   return await httpService({
-    url: `${apiBasePath}/fileupload/import${paramString}`,
+    url: `/api/fileupload/import`,
     method: 'POST',
+    ...(query ? { query }: {}),
```

Comment:
We were appending this to the request string. Add here as a query instead when it exists